File: swarmmanagercontroller/src/main/java/co/uk/swarmbit/docker/api/secrets/parameters/SecretsListParameters.java
package co.uk.swarmbit.docker.api.secrets.parameters;

import co.uk.swarmbit.rest.QueryParam;
import co.uk.swarmbit.util.EncoderDecoder;
import co.uk.swarmbit.docker.api.common.parameters.FiltersParameters;

import java.util.Optional;

public class SecretsListParameters implements FiltersParameters {

    private static final String FILTERS_NAME = "filters";

    private Optional<QueryParam> filters;

    public SecretsListParameters() {
        filters = Optional.empty();
    }

    public Optional<QueryParam> getFilters() {
        return filters;
    }

    public SecretsListParameters setFilters(SecretsFilters filters) {
        if (filters != null) {
            this.filters = Optional.of(new QueryParam(FILTERS_NAME, EncoderDecoder.jsonUrlEncode(filters.getJson())));
        }
        return this;
    }
}

File: swarmmanagercontroller-new/src/main/kotlin/com/swarmbit/docker/api/secrets/parameters/SecretsListParameters.kt
package com.swarmbit.docker.api.secrets.parameters

import com.swarmbit.docker.api.common.parameters.FiltersParameters
import com.swarmbit.docker.api.common.rest.model.QueryParam

class SecretsListParameters : FiltersParameters {
    companion object {
        private const val FILTERS_NAME = "filters"
    }

    private var innerFilters: QueryParam? = null

    fun setFilters(filters: SecretsFilters): SecretsListParameters {
        this.innerFilters = QueryParam(FILTERS_NAME, filters.asUrlJson())
        return this
    }

    override val filters: QueryParam?
        get() = innerFilters
}
